mdmonitor.c

```diff
@@ -639,11 +639,19 @@ static void execute_alert_cmd(const struct event_data *data)
  */
 static void send_event_email(const struct event_data *data)
 {
-	FILE *mp, *mdstat;
+	FILE *mp = NULL, *mdstat;
 	char buf[BUFSIZ];
 	int n;
 
-	mp = popen(Sendmail, "w");
+	if (info.mailfrom) {
+		char *cmd = NULL;
+
+		if (xasprintf(&cmd, "%s -f%s", Sendmail, info.mailfrom) > 0)
```

Comment:
I don't see a reason to not used simple snprintf instead. Do I miss something?

Reply:
It requires using a fixed-size buffer on the stack. I can do that of course. I was just using a coding pattern that I'd observed in existing mdadm code.

Reply:
This one might be reported as issue by static analysis, we don't need anyone to think about it.

Please rework it if it is not a big deal :)

Reply:
Right, it's not a big deal. But I wouldn't expect static checkers to complain about this code. Anyway, will do.